From 58a13a7c3610b3ecb6f2eed7480b567abf4444b6 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 2 Jan 2020 17:49:25 +0000
Subject: [PATCH] screenInfoFromQScreen: fallback on primary screen if no
 screen

This way, a correct value can be retrieved for the size of the screen.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/render_widget_host_view_qt.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/core/render_widget_host_view_qt.cpp b/src/core/render_widget_host_view_qt.cpp
index 3be0c939..e586b673 100644
--- a/src/core/render_widget_host_view_qt.cpp
+++ b/src/core/render_widget_host_view_qt.cpp
@@ -255,6 +255,11 @@ private:
 static content::ScreenInfo screenInfoFromQScreen(QScreen *screen)
 {
     content::ScreenInfo r;
+    
+    if (!screen) {
+        screen = qGuiApp->primaryScreen();
+    }
+    
     if (screen) {
         r.device_scale_factor = screen->devicePixelRatio();
         r.depth_per_component = 8;
-- 
2.17.0

